src/storage/node-fs.ts

```diff
@@ -1,54 +1,71 @@
-import type { Stats } from "node:fs";
 import { resolve, parse, join } from "pathe";
 import { createError } from "h3";
 import { cachedPromise, getEnv } from "../utils";
 import type { IPXStorage } from "../types";
 
 export type NodeFSSOptions = {
-  dir?: string;
+  dir?: string | string[];
   maxAge?: number;
 };
 
+function resolveDirs(options: NodeFSSOptions) {
+  if (!options.dir || !Array.isArray(options.dir)) {
+    const dir = resolve(options.dir || getEnv("IPX_FS_DIR") || ".");
+    return [dir];
+  }
+
+  return options.dir.map((dir) => {
+    return resolve(dir);
+  });
+}
+
 export function ipxFSStorage(_options: NodeFSSOptions = {}): IPXStorage {
-  const rootDir = resolve(_options.dir || getEnv("IPX_FS_DIR") || ".");
+  const dirs = resolveDirs(_options);
   const maxAge = _options.maxAge || getEnv("IPX_FS_MAX_AGE");
 
-  const _resolve = (id: string) => {
-    const resolved = join(rootDir, id);
-    if (!isValidPath(resolved) || !resolved.startsWith(rootDir)) {
+  const _getFS = cachedPromise(() => import("node:fs/promises"));
+
+  const getMeta = async (id: string) => {
+    const errors = new Set<string>();
+
+    for (const dir of dirs) {
+      const filePath = join(dir, id);
+
+      if (!isValidPath(filePath) || !filePath.startsWith(dir)) {
+        errors.add("IPX_FORBIDDEN_PATH");
+      }
+
+      try {
+        const fs = await _getFS();
+        const stats = await fs.stat(filePath);
+        return { stats, filePath };
+      } catch (error: any) {
+        errors.add(
+          error.code === "ENOENT" ? "IPX_FILE_NOT_FOUND" : "IPX_FORBIDDEN_FILE",
```

Comment:
In case of `ENOENT` we can simply ignore loop (if finally there was no results, we throw `IPX_FILE_NOT_FOUND`)

Otherwise we can immediately throw `IPX_FORBIDDEN_FILE` and stop searching

(this way we don't need `error` array too)

Reply:
I'd argue we need it still, because a path may be forbidden within one directory, but be later found in another. I screwed up with `IPX_FORBIDDEN_PATH` though, fixing it now :)

Reply:
Fixed. Originally I had it implemented without the `error` array. If it is safe to assume `IPX_FORBIDDEN_PATH` will persist across all dirs we can simplify the array to boolean flag. But `IPX_FORBIDDEN_FILE` can be easily thrown by a path resolved in one dir and be found in another dir.

Not sure how to write a test for this one, as we need to fake forbidden access.
```
chmod = getChmod(path)
updateChmod(path, 000) // ???
// do test
updateChmod(path, chmod)
```

Reply:
security-wise caling stat unnecessarily is not the best idea. will push refactor to early. If a path exists and is inaccessible, it is either wrong server setup or something wrong. we should stop asap.